### DISCUSSION TITLE
Best way to add custom terminal output at will?

### DISCUSSION THREAD

**Discussion by Ksauder** [pytest-dev/pytest#8980] [Q&A]
I'm working on a custom plugin and I'd like to print certain important status messages out to the terminal from hooks and fixtures. I am currently grabbing the terminalreporter plugin and using write_line(), which seems to work sometimes and not others. I've also tried grabbing the capturemanager and calling my write_line() method within a global_and_fixture_disabled() context. This doesn't work if I put the context within my wrapper method for write_line, but does work if I use the context directly within the hookimpl I'm working in.

How should this work in this scenario:

```
custom_fixture():
    yield
    config.hook.custom_teardown_hook()

custom_teardown_hook():
    print_uncaptured_output()
```

Willing to accept the answer: 'you're doing everything wrong'.

**Comment by nicoddemus** [comment]
Hi @Ksauder,

> grabbing the terminalreporter plugin and using write_line()

That's what I would do, and should work every time except in rare cases I think.

**Reply by Ksauder** [reply]
Then perhaps I have found a rare case. I will demo this further given my fairly large plugin and see what I can find. So far it has been flaky as to when it actually lets the output through.

**Reply by Ksauder** [reply]
Turns out it was a docker issue. I run the test framework in a docker container and use a script to run the docker with proper volumes and args. It worked 99% of the time, but when trying to manipulate the CaptureManager, for some reason, I wouldn't get most of the output except the last calls output in a loop test. docker run with a -t fixes it.
